---
title: "Disorder in Cdk1/2 targets"
output:
  html_document:
    df_print: paged
    fig_caption: yes
  pdf_document: default
params:
  d: !r Sys.Date()
bibliography: references.bib
---
updated: `r params$d`  

```{r setup,echo=FALSE,message=FALSE,warning=FALSE,prompt=FALSE}
library(jsonlite)
library(readr)
library(ggplot2)
library(GGally)
library(vcd)
library(kableExtra)
library(ggpubr)
library(Rcpi)
#if fails => sudo R CMD javareconf; reinstall; reboot R
library(Biostrings)
library(eulerr)
library(flextable)
library(stringr)
library(DT)
library(dplyr)

source("getDisorder.R")
source("contingencyTablesAnalysis.R")


#################################
######## Disorder data ##########
#################################


# ___________MOBI DB_____________

# yeastDiso is a complex database from MobiDB Saccharomyces cerevisiae. 

yeastDiso<-stream_in(con = file("disorder_UP000002311.mjson"),verbose = F)
yeastDiso$mobidb_consensus <- yeastDiso$mobidb_consensus$disorder
colnames(yeastDiso)[5] <- "disorder"


# ___________ SPOT _____________
spotDisorderList <- getSpotDisoMatrix("spot_run_yeast/")


yeastDiso$length <- nchar(yeastDiso$sequence)
yeastDiso$predDisoPerc <- getPredDiso(yeastDiso,output = "percentage")
yeastDiso$predLiteDisoPerc <- getPredLiteDiso(yeastDiso,output = "percentage")
yeastDiso$derivedDisoPerc <- getDerivedDiso(yeastDiso,output = "percentage")
yeastDiso$dbDisoPerc <- getDBDiso(yeastDiso,output = "percentage")
yeastDiso$predSpotDisoPerc <- getSpotPredDiso(yeastDiso,output = "percentage","acc")



#################################
###### CDK1 targets data ########
#################################


# ___________Ubersax data_____________
Ubersax_data <- read_delim("CDK1targetSacCer.csv", 
                          "\t", quote = "\\\"", escape_double = FALSE, 
                          col_types = cols(X1 = col_skip(), input = col_skip(), 
                          `yourlist:M201907046746803381A1F0E0DB47453E0216320D4B0BD1N` = col_skip()), 
                          trim_ws = TRUE)

# Now the comumn "disorder" is a unnamed list with a complex structure for each row of the data frame

#The row 294 cannot be mapped with the uniprot, I will remove it from CDK1targetsSacCer and yeastDisoCDK1target. Slect only the rows wuth P-score > 2 most likely targets based on the paper

Ubersax_data<-subset(Ubersax_data[-294,])
yeastDiso_Ubersax_subset<-yeastDiso[match(Ubersax_data$Entry,yeastDiso$acc),]
Ubersax_data$length <- nchar(yeastDiso_Ubersax_subset$sequence)
Ubersax_data$predDisoPerc <- yeastDiso_Ubersax_subset$predDisoPerc
Ubersax_data$predLiteDisoPerc <- yeastDiso_Ubersax_subset$predLiteDisoPerc
Ubersax_data$derivedDisoPerc <- yeastDiso_Ubersax_subset$derivedDisoPerc
Ubersax_data$dbDisoPerc <- yeastDiso_Ubersax_subset$dbDisoPerc
Ubersax_data$predSpotDisoPerc <- yeastDiso_Ubersax_subset$predSpotDisoPerc
# CDK1targetSacCer<-subset(CDK1targetSacCer[-294,])
# yeastDisoCDK1target<-yeastDiso[match(CDK1targetSacCer$Entry,yeastDiso$acc),]
# CDK1targetSacCer$length <- nchar(yeastDisoCDK1target$sequence)


# Ubersax_data$target<-cut(Ubersax_data$`P-Score`,breaks = c(2),labels = c("Non Cdk1 target","Cdk1 target"))
Ubersax_data$MW_kDa<-(Ubersax_data$length*120)/1000
Ubersax_data$nmol<- (Ubersax_data$`Protein Level`/Ubersax_data$MW_kDa)/1000
Ubersax_data$nEq_STP<- Ubersax_data$nmol*Ubersax_data$`[ST]-P`
Ubersax_data$nEq_STPXKR<- Ubersax_data$nmol*Ubersax_data$`[ST]-P-X-[KR]`
Ubersax_data$target<-cut(Ubersax_data$`P-Score`,breaks = c(-Inf,2,Inf),labels = c("Non Cdk1 target","Cdk1 target"),right = F)
Ubersax_data$targetStrict<-cut(Ubersax_data$`P-Score`,breaks = c(-Inf,3,Inf),labels = c("Non Cdk1 target","Cdk1 target"),right = F)



# ___________Holt Data_____________
uniprot_yeast <- read_delim("uniprot_yeast.tab",
"\t", escape_double = FALSE, trim_ws = TRUE)
Holt_MS <- read_delim("Holt_CDK1targetSacCer.tsv",
"\t", escape_double = FALSE, trim_ws = TRUE)

#Remove phosphoTyrosines
Holt_MS<-subset(Holt_MS,!substr(site,start = 1,stop = 1)=="Y")
#Add Uniprot accesion. 58 entries lost (out of 9067) due to ID conversion
Holt_MS <- Holt_MS[which(Holt_MS$ORF %in% uniprot_yeast$OLN),]

Holt_MS$Uniprot <- uniprot_yeast$`Swiss-Prot-AC`[base::match(Holt_MS$ORF,uniprot_yeast$OLN)]


#Here I'll select all the phosphorylated S and T under the autors considerations.  

Holt_MS_Cdk1_all_phosphosites <- Holt_MS %>% dplyr::filter(`log2H/L_asynchronous`<(-1)|`log2H/L_nocodazole`<(-1)|`log2H/L_clb2`<(-1))

#select all CDK targets (genes with at least one phosphosite experimentallly detected)

Holt_cdkAllTargetList <- unique(Holt_MS_Cdk1_all_phosphosites$Uniprot)
Holt_nonCdkAllTargetList <- unique(Holt_MS[!(Holt_MS$Uniprot %in% Holt_MS_Cdk1_all_phosphosites$Uniprot),]$Uniprot)

#generate a data.frame with two columns: uniprpot and target. Merge with the proteins info extracted from yeastDiso.

Holt_data <- as.data.frame(rbind(cbind(as.character(Holt_cdkAllTargetList),rep("Cdk1 target",length(Holt_cdkAllTargetList))),cbind(as.character(Holt_nonCdkAllTargetList),rep("Non Cdk1 target",length(Holt_nonCdkAllTargetList)))))

colnames(Holt_data) <- c("Uniprot","target_all")
Holt_data$Uniprot <- as.character(Holt_data$Uniprot)

Holt_data<-merge.data.frame(Holt_data,yeastDiso,by.x = "Uniprot",by.y = "acc")

Holt_data$target_all <- relevel(factor(Holt_data$target_all),"Non Cdk1 target")
# Anotate only the phosphosites that respect the minimal consensus motif

Holt_MS_Cdk1_consensus_phosphosites <- Holt_MS %>% dplyr::filter(grepl("[st]P",centered_15mer)) %>% dplyr::filter(`log2H/L_asynchronous`<(-1)|`log2H/L_nocodazole`<(-1)|`log2H/L_clb2`<(-1))

Holt_cdkConsensusTargetList <- unique(Holt_MS_Cdk1_consensus_phosphosites$Uniprot)
Holt_nonCdkConsensusTargetList <- unique(Holt_MS[!(Holt_MS$Uniprot %in% Holt_MS_Cdk1_consensus_phosphosites$Uniprot),]$Uniprot)

Holt_consensus <- as.data.frame(rbind(cbind(as.character(Holt_cdkConsensusTargetList),rep("Cdk1 target",length(Holt_cdkConsensusTargetList))),cbind(as.character(Holt_nonCdkConsensusTargetList),rep("Non Cdk1 target",length(Holt_nonCdkConsensusTargetList)))))

colnames(Holt_consensus) <- c("Uniprot","target_consensus")
Holt_consensus$Uniprot <- as.character(Holt_consensus$Uniprot)

Holt_data<-merge.data.frame(Holt_data,Holt_consensus,by = "Uniprot")

Holt_data$target_consensus <- relevel(factor(Holt_data$target_consensus),"Non Cdk1 target")

# # adding psites
# 
# holt_phosphosites <-aggregate(substring(site,2,1000) ~ Uniprot, data = Holt_MS_Cdk1_all_phosphosites, paste, collapse=",")
# colnames(holt_phosphosites) <- c("Uniprot","sites")
# # Holt_MS_Cdk1_all_phosphosites has only the Psites with fold Chnge>2 but Holtdata has all the proteins
# Holt_data$phosphosites <- holt_phosphosites$sites[match(Holt_MS_Cdk1_all_phosphosites$Uniprot,holt_phosphosites$Uniprot)]



# ___________Phosphosite Human Data_____________
# PSP Search Result Legend	 
# COLUMN HEADER	DESCRIPTION
# PROTEIN	Primary name in PhosphoSitePlusÂ® (PSP)
# GENE	Unique ID (UID) assigned to a gene by the HUGO Nomenclature Committee and adapted by HUPO for the cognate protein.
# ORGANISM	The species from which either the kinase and/or substrate proteins have been derived
# ACC_ID	Primary accession ID in PSP
# MW_(DA)	Molecular weight in Daltons.
# MOD_RSD	The location within the protein sequence of the amino acid residue that is posttranslationally modified (modsite).
# SITE_+/-7_AA	The modsite plus flanking sequence (+/- 7 rsds.). The modsite, as well as other residues within the flanking sequence that are known to be posttranslationally modified, are displayed in lower-case letters.
# SITE_GRP_ID	Unique identifier of the modification site and its homologous sites in all proteoforms and species
# DOMAIN	The Pfam residue in which the modsite is located.
# METHOD	The experimental method in which kinase-substrate relationships were reported. Vivo = in vivo: determined from reactions within living cells, cell cultures or organisms. Vitro = in vitro: determined from reactions outside of living cellular structures.
# LTP_LIT	The number of literature records using low-throughput (LTP) experimental techniques in which the modsite has been reported. LTP results may be more reliable than MS2 results.
# MS2_LIT	The number of literature records using tandem mass spectrometry (MS2) experimental techniques in which the modsite has been observed.
# MS2_CST	The number of curation sets (CS) derived from MS2 experiments performed at Cell Signaling Technology (CST) in which the modsite has been observed.
# CST_CAT#	The catalog number(s) of CST antibodies specific for the associated modsite.

#there's overlapping between the targets. I create 3 differents DF and I will merge them

phosphosite_CDK1Target_HS <- read_delim("phosphosite_CDK1Target_HS.tab","\t", escape_double = FALSE, trim_ws = TRUE)
phosphosite_CDK2Target_HS <- read_delim("phosphosite_CDK2Target_HS.tab","\t", escape_double = FALSE, trim_ws = TRUE)
phosphosite_CDK3Target_HS <- read_delim("phosphosite_CDK3Target_HS.tab","\t", escape_double = FALSE, trim_ws = TRUE)

phosphosite_CDKFamilyTarget_HS <- merge.data.frame(phosphosite_CDK1Target_HS,phosphosite_CDK2Target_HS,by = c("GENE","PROTEIN","ACC#","MOD_RSD"),all = T)
colnames(phosphosite_CDKFamilyTarget_HS)[-(1:4)]<-c("METHOD.CDK1","LTP_LIT.CDK1","MS2_LIT.CDK1","MS2_CST.CDK1","Kinase.CDK1","METHOD.CDK2","LTP_LIT.CDK2","MS2_LIT.CDK2","MS2_CST.CDK2","Kinase.CDK2")
phosphosite_CDKFamilyTarget_HS <- merge.data.frame(phosphosite_CDKFamilyTarget_HS,phosphosite_CDK3Target_HS,by = c("GENE","PROTEIN","ACC#","MOD_RSD"),all = T)
colnames(phosphosite_CDKFamilyTarget_HS)[-(1:14)]<-c("METHOD.CDK3","LTP_LIT.CDK3","MS2_LIT.CDK3","MS2_CST.CDK3","Kinase.CDK3")


```


# Protein Disorder

## MobiDB
  
The [MobiDB](http://mobidb.bio.unipd.it) is database of protein disorder and mobility annotations. Several curated datasets for intrinsic disorder and folding upon binding have been integrated from specialized databases. The indirect evidence has also been expanded to better capture information available in the PDB, such as high-temperature residues in X-ray structures and overall conformational diversity.  
The database is organized in three layers containing data of different nature: Disorder, Linear Interacting Peptides (LIPs) and Dynamic structure. The "Disorder" database bears data about protein regions without a defined tri-dimensional structure. The "LIPs" layer contains information about regions or residues that interact with other proteins or DNA/RNA preserving a linear (disordered) structure. At last, the data hosted in the "Dynamic structure" layer describes the propensity of a residue to assume a specific secondary structure conformation. When extracted from chemical shifts can be interpreted as a measure of the secondary structure populations in solution.
Each one of three layers features three quality levels of annotation from high to low quality (Fig. 1), presenting different coverage of the universe of known proteins.
  
```{r confidence pyramid ,echo=FALSE,message=FALSE,warning=FALSE,fig.cap="Figure 1: MobiDB confidence vs. coverage. DB refers to manually curated annotations from external databases. Indirect consist in annotations derived/calculated information from experimental data, i.e. PDB structures and/or chemical shifts. Predicted annotations are made using exclusively bioinformatic tools", out.width = "400px"}

knitr::include_graphics("/home/gdubra/projects/cdk1_cyclinB1/phase_separation/disorder/images/mobiDB_confVScoverage.png")

```  


### Protein disorder  
  
As a first approach, the Mobidb database was explored for the whole annotated proteome of Saccharomyces cerevisiae. The data set comprehends the information of disordered region over 6721 proteins. As explained before, each entry has annotations of disorder in three different levels (DB, indirect, Predicted). At each one of those levels, information from several sources is annotated in order to reach to a consensus of the regions with or without a defined tri-dimensional structure. For the disorder predictors annotated in MobiDB the regions with a predicted disorder regions are marked with a "D". For those levels in which a consensus is reached ("DB" and "Derived") the regions are marked as "S" for structured, "C" for conflicts between different the methods, and "D" for disorder.  

```{r all data summary ,echo=FALSE,message=FALSE,warning=FALSE}
npredictors=0
nmobidblite=0
nsimple=0
#methods: mobidb-lite, simple
for (i in 1:nrow(yeastDiso)) {
  if(!is.null(yeastDiso[i,5]$predictors[[1]])){
    npredictors=npredictors+1
    }
  if( "mobidb-lite" %in% yeastDiso[i,5]$predictors[[1]]$method){
    nmobidblite=nmobidblite+1
    }
  if( "simple" %in% yeastDiso[i,5]$predictors[[1]]$method){
    nsimple=nsimple+1
    }  
  }


#methods: full bfactor missing_residues mobile
nderived=0
nfull=0
nbfactor=0
nmissing_residues=0
nmobile=0

for (i in 1:nrow(yeastDiso)) {
  if(!is.null(yeastDiso[i,5]$derived[[1]])){
    nderived=nderived+1
  }
  if( "full" %in% yeastDiso[i,5]$derived[[1]]$method){
    nfull=nfull+1
  }
  if( "bfactor" %in% yeastDiso[i,5]$derived[[1]]$method){
    nbfactor=nbfactor+1
  }
  if( "missing_residues" %in% yeastDiso[i,5]$derived[[1]]$method){
    nmissing_residues=nmissing_residues+1
  }
  if( "mobile" %in% yeastDiso[i,5]$derived[[1]]$method){
    nmobile=nmobile+1
  }
}
#methods: full
ndb=0
for (i in 1:nrow(yeastDiso)) {
  if(!is.null(yeastDiso[i,5]$db[[1]])){
    ndb=ndb+1
    }
  }
```

In order to understand the data the first analysis to do is to evaluate the coverage of the different levels of annotations in the database. In this way, a level or a combination of them can be selected for assigning the disorder regions to each protein considering the compromise between the quality of the annotation and the coverage of the protein set of interest.  
DB is the level with less coverage, since its annotations are manually curated, and it has **`r ndb`** entries covering a **`r round((ndb/nrow(yeastDiso))*100,2)`%** of the whole yeast proteome. The data annotated in the "Derived" level holds **`r nderived`** entries and this covers **`r round((nderived/nrow(yeastDiso))*100,2)`%** of available proteins for S. cerevisiae. Since the predictors do not have any limitations in terms of which proteins could these methods be applied to, they cover the entire protein set.  
  
| Level      | Method           | N° entries            | Level coverage (%)                            | Method coverage (%)                                 |
|------------|------------------|-----------------------|-----------------------------------------------|-----------------------------------------------------|
| DB         | full             | `r ndb`               | `r round((ndb/ndb)*100,2)`                    | `r round((ndb/nrow(yeastDiso))*100,2)`              |
| Derived    | full             | `r nderived`          | `r round((nderived/nderived)*100,2)`          | `r round((nderived/nrow(yeastDiso))*100,2)`         |
|            | bfactor          | `r nbfactor`          | `r round((nbfactor/nderived)*100,2)`          | `r round((nbfactor/nrow(yeastDiso))*100,2)`         |
|            | missing-residues | `r nmissing_residues` | `r round((nmissing_residues/nderived)*100,2)` | `r round((nmissing_residues/nrow(yeastDiso))*100,2)`|
|            | mobile           | `r nmobile`           | `r round((nmobile/nderived)*100,2)`           | `r round((nmobile/nrow(yeastDiso))*100,2)`          |
| Predictors | simple           | `r nsimple`           | `r round((nsimple/nsimple)*100,2)`            | `r round((nsimple/nrow(yeastDiso))*100,2)`          |
|            | mobidb-lite      | `r nmobidblite`       | `r round((nmobidblite/nsimple)*100,2)`        | `r round((nmobidblite/nrow(yeastDiso))*100,2)`      |
Table 1:  Percent of covered entries by each Level and method of the MobiDB database.  

## SPOT disorder prediction  
  
For predicting IDRP common deep learning techniques, such as recurrent neural networks and window-based artificial neural networks, while effective at propagating local errors within sequence neighbors, are ineffective at modeling long-range (non-local) interactions between amino acid residues that are structural but not sequence neighbors. Because residue–residue interactions are dominated by structural neighbors, how to account for them is the key for improving sequence-based prediction of protein structural and functional properties.
The long-range dependence between a series of time-resolved events can be better captured by enforcing the constant error flow so that useful long-range interactions can be memorized. 
SPOT (Sequence-based Prediction Online Tools for disorder) captures nonlocal interactions that are essential for determining whether a protein will fold (structured) or will not fold (intrinsically disordered) into a unique three-dimensional structure, by employing deep bidirectional LSTM cells. The bidirectional network will allow us to capture both forward and backward information contained in protein sequences. This new method is found to be highly effective in predicting both short and long disordered regions without separated training, despite disordered regions of different sizes having different compositions of amino acids

The evolutionary content is established through the use of a Position-Specific Scoring Matrix (PSSM), generated by three iterations of the PSI-BLAST algorithm (Altschul et al., 1997) against the NCBI’s Non-Redundant (NR) sequence database for each protein. The Shannon entropy is also calculated to represent the information content in these probabilities per residue (Shannon, 1948). The average Shannon entropy over the entire protein is also utilized as an input feature of the general conservation of the whole protein. This leads to a total of 22 evolutionary features used for prediction.
We also obtained 17 predicted structural features from the SPIDER2 predictor (1 ASA, 1 CN, 4 HSE based on Cα and Cβ atoms, respectively and the 11 predicted SS and sine/cosine of the backbone angle values) (Heffernan et al., 2015, 2016). Finally, seven commonly-used physicochemical properties, including hydrophobicity and polarizability, are used as features as provided by Meiler et al. (2001).

Thus, using these features resulted in a 46-length feature vector for each amino acid. 
[@Hanson2016]
  
## Disorder statistics

The raw data collected from the MobiDB and the SPOT run for the whole yeast genome should be adjusted and transformed in order to have quantitative measurements of protein disorder that can be compared between the different methods. The processing of the MobiDB database present a bigger challenge since the database is designed as relational and the files fetched (mjson format) when transformed into a flat table format generate columns with nested matrices as values.
For the SPOT disorder prediction, all the proteins in S. cerevisiae were input into a batch software run, and all IDR estimated were annotated in one file per protein. All the files were scanned with an ad-hoc defined function and the data was added to the disorder tables generated by the parsing of the MobiDB information.

### Percentage of disordered aminoacids  
  
  
Once we have established the disorder regions for each protein it is useful to have a metric describing the global disorder status for each entry in order to summarise in one number the degree of disorder and to being able to make comparisons at the protein level. The percentage (or ratio) of disorder amino acids over the total number of amino acids of a protein constitute the most direct metric, providing one value that we can assign to each proteins describing its state of disorder.



```{r histograms predictors,echo=FALSE,message=FALSE,warning=FALSE, fig.cap="Figure 2: Histogram for the disorder percentages estimated by the two predictors"}

ggplot(data = yeastDiso)+geom_density(aes(x=predLiteDisoPerc,fill="MobiDB lite"),bins = 30,alpha=0.5,position = "identity")+geom_density(aes(x=predDisoPerc,fill="Simple"),bins = 30,alpha=0.5,position = "identity")+ geom_density(aes(x=predSpotDisoPerc,fill="SPOT"),bins = 30,alpha=0.5,position = "identity") + theme_minimal() + xlab("% of disorder residues") + scale_fill_manual(name = "Predictor", values = c("Simple"="darkorange1","MobiDB lite" = "chartreuse3", "SPOT" = "firebrick1"))
```
  
  
Both of the predictors featured by the MobiDB database have a notorious discrepancy on the number of residues on which they assign the "Disorder" status. This could be due, in principle, to the differences in their conceptions.  The "Simple" method is basically consensus derived from some of the already established disorder bioinformatic predictors and features region-wise annotations. In contrast, the method called MobiDB lite is, in fact, an *ad hoc* development that not only uses a consensus between these methods but also quantifies the agreement among different predictors, and it is optimized for detecting long intrinsically disordered regions (IDR) and the detection of short regions are filtered out. This could explain why Figure 1 presents a great proportion of proteins without any disorder residue for the MobiDB lite predictor. It is understandable then, that the ratio of the residues detected as intrinsically disorder over the total length of the proteins, by those different methods, are not in agreement and even they seem to be complementary between them.  
  
For the "Derived" and "DB" levels, the identification of IDR has, allegedly, a higher degree of confidence. The comparison the percentage of disorder residues estimated by those levels (Figure 3) and the prediction methods could help to clarify which prediction is the better option if we want to use the percentage of disordered amino acids as metric to evaluate disorder at the protein level.  
  
  
```{r histograms dbANDderived,echo=FALSE,message=FALSE,warning=FALSE, fig.cap="Figure 3: Histogram for the disorder percentages estimated for the 'DB' and 'Derived' levels"}

ggplot(data = yeastDiso)+geom_density(aes(x=dbDisoPerc,fill="DB"),bins = 30,alpha=0.5,position = "identity")+geom_density(aes(x=derivedDisoPerc,fill="Derived"),bins = 30,alpha=0.5,position = "identity") + theme_minimal() + xlab("% of disorder residues") + scale_fill_manual(name = "Method", values = c("Derived"="darkgoldenrod1","DB" = "deepskyblue2"))
```
  
The use of the derived data for assigning IDRs seems to present a distribution of the percentage of disorder residues congruent with the MobiDB lite predictor, presenting a considerable proportion of protein not having any disorder residues.

Similar distributions of the percentage of disorder residues, for two different predictors, could be validated if the predictors are assigning similar percentage values to the same proteins. If this is the case, we should be able to find a correlation between the values calculated by different methods. In Figure 4, pairwise comparisons for both of the prediction methods and the Derived and DB annotations are shown, together with the density distributions and the corresponding correlation coefficients. Since we are interested in elucidating how CDK phosphorylation could be involved in the disorder states of its targets, another level of complexity is added to this plot. The percentage of phosphorylatable residues ([ST]P) were calculated for all the proteins and Low, Mid and High levels of putative phosphorylations sites were defined with arbitrary breaks (0-15%, 15-20%,20-60% ) of the percentage values, based on the quartiles of the distribution.

```{r comparison scatters ,echo=FALSE,message=FALSE,warning=FALSE, fig.cap="Figure 4: Pairwise comparison of the % of disorder residues determined by the different methods",fig.height=8,fig.width=10 }

compDataSet <- yeastDiso[,c("sequence","predDisoPerc","predLiteDisoPerc","predSpotDisoPerc","derivedDisoPerc","dbDisoPerc")]
phosphorylatableResidues <- numeric()
for (phosoSites in gregexpr(pattern = "[ST]P",as.character(compDataSet[,"sequence"]))){phosphorylatableResidues<-c(phosphorylatableResidues,length(phosoSites))}
compDataSet$nPhosphoRes <- phosphorylatableResidues
compDataSet[,"sequence"] <- nchar(as.character(compDataSet[,"sequence"]))
colnames(compDataSet)[1] <- "proteinLength"
compDataSet$percPhosphoRes <- (compDataSet$nPhosphoRes/compDataSet$proteinLength)*100
compDataSet$levelPhosphoRes<-cut(compDataSet$percPhosphoRes,breaks = c(0,0.45,1,10),labels = c("Low","Mid","High"))

# print(summary(compDataSet$percPhosphoRes))
GGally::ggpairs(data = compDataSet,columns = c(2:6), ggplot2::aes(colour=levelPhosphoRes,alpha=0.5),xlab="% of disorder residues", ylab="% of disorder residues",columnLabels=c("Simple","Lite","SPOT","Derived","DB")) +
     theme(panel.grid.minor = element_blank(),panel.grid.major = element_blank()) + theme_classic()

```
  
  
#### Lenght distribution in the detection of IDRs 
  

None of the methods for estimating the disorder regions of proteins seem to be in agreement with the other methods in the percentage of disorder residues for a given protein. The correlation coefficients are low, and the ratio of putative phosphorylation sites does not have any significant impact on these calculations. Moreover, the distribution of the percentage of phosphorylatable residues appears to be completely random in all the pairwise comparison scatterplots. 
The lack of correlation in this metric for different methods does not directly imply that the methods are not identifying the same regions. In fact, all of the aminoacids classified as disordered by the MobiDB lite method, for all the proteins, were also detected by the simple method. Disparities in the lengths of the disorder stretches lead to these observed effects.  

```{r stretches of disorder ,echo=FALSE,message=FALSE,warning=FALSE, fig.cap="Figure 5:distribution of the lenghts of disorder stretches" }

predStretchDist <- getPredDiso(yeastDiso,"lenghts")
predStretchDist <- cbind(predStretchDist,rep("Simple predictor",length(predStretchDist)))
predLiteStretchDist <- getPredLiteDiso(yeastDiso,"lenghts")
predLiteStretchDist <- cbind(predLiteStretchDist,rep("MobiDB-lite predictor",length(predLiteStretchDist)))
predSpotStretchDist <- getSpotPredDiso(yeastDiso,"lenghts","acc")
predSpotStretchDist <- cbind(predSpotStretchDist,rep("SPOT predictor",length(predSpotStretchDist)))
derivedStretchDist <- getDerivedDiso(yeastDiso,"lenghts")
derivedStretchDist <- cbind(derivedStretchDist,rep("Derived",length(derivedStretchDist)))
dbStretchDist <- getDBDiso(yeastDiso,"lenghts")
dbStretchDist <- cbind(dbStretchDist,rep("DB",length(dbStretchDist)))





stretchData <-rbind(predStretchDist,predLiteStretchDist,predSpotStretchDist,dbStretchDist,derivedStretchDist)
stretchData <-as.data.frame(stretchData)
colnames(stretchData) <- c("stretchLength","method")
stretchData$method <- as.factor(stretchData$method)
stretchData$method <- factor(stretchData$method, levels = c("Simple predictor","MobiDB-lite predictor","SPOT predictor","Derived","DB"))
stretchData$stretchLength <- as.numeric(stretchData$stretchLength)




ggplot(data = stretchData,aes(x=method,y=stretchLength)) + geom_violin(aes(fill=method), alpha = 0.7,show.legend = FALSE,bw=40) +
    theme_minimal() +
    geom_boxplot(width=0.05) +
    scale_fill_manual(values = c("Simple predictor"="darkorange1","MobiDB-lite predictor" = "chartreuse3","SPOT predictor" = "firebrick1","Derived"="darkgoldenrod1","DB" = "deepskyblue2")) +
    xlab("Method") + ylab("Lenght of IDR") +
    scale_y_continuous(limits = c(0,600))
```
  

```{r compare index ,echo=FALSE,message=FALSE,warning=FALSE}

addUniprotToIdx <- function(df,disoIdx){
  aux_disoIdx <- disoIdx
  for (i in 1:nrow(df)) {
    aux_disoIdx[[i]] <- paste(df$acc[[i]],aux_disoIdx[[i]],sep = "_")
  }
  return(unlist(aux_disoIdx))
}

simpleDisoIdx <- getPredDiso(yeastDiso,"indices")
simpleDisoIdx <- addUniprotToIdx(yeastDiso,simpleDisoIdx)
liteDisoIdx <- getPredLiteDiso(yeastDiso,"indices")
liteDisoIdx <- addUniprotToIdx(yeastDiso,liteDisoIdx)
spotDisoIdx <- getSpotPredDiso(yeastDiso,"indices","acc")
spotDisoIdx <- addUniprotToIdx(yeastDiso,spotDisoIdx)
derivedDisoIdx <- getDerivedDiso(yeastDiso,"indices")
derivedDisoIdx <- addUniprotToIdx(yeastDiso,derivedDisoIdx)
dbDisoIdx <- getDBDiso(yeastDiso,"indices")
dbDisoIdx <- addUniprotToIdx(yeastDiso,dbDisoIdx)


sets <- list("Simple predictor"=simpleDisoIdx,"Lite predictor"=liteDisoIdx,"SPOT predictor"=spotDisoIdx)

eulerr_options(fills=list(fill=c("darkorange1","chartreuse3","firebrick1"),alpha=0.4))
plot(euler(sets, shape = "ellipse"), quantities = TRUE)


sets <- list("SPOT predictor"=spotDisoIdx,"Derived"=derivedDisoIdx,"DB"=dbDisoIdx)

eulerr_options(fills=list(fill=c("firebrick1","darkgoldenrod1","deepskyblue2"),alpha=0.4))
plot(euler(sets, shape = "ellipse"), quantities = TRUE)


# percCovLiteInSimple <- numeric()
# for (i in 1:nrow(yeastDiso)) {
#   liteIndex<-which(yeastDiso$disorder$predictors[[i]]$method=="mobidb-lite")
#   simpleIndex<-which(yeastDiso$disorder$predictors[[i]]$method=="simple")
#   if (length(liteIndex)!=0) {
#     matLite<-yeastDiso$disorder$predictors[[i]]$regions[[liteIndex]]
#     matSimple<-yeastDiso$disorder$predictors[[i]]$regions[[simpleIndex]]
#     if (class(matLite)=="matrix") {
# #getting only the disordered rows. Is really important to reconvert to a matrix of ncol=3 just in case I index only one row and I get a vector. the the subindex mat[,2] will throw incorrect number of dimensions
#       # mat<-matrix(mat[mat[,3]=="D",],ncol=3)
#       litePos <-numeric()
#       litePos <-apply(matLite, 1, function(x){litePos<-c(litePos,x[1]:x[2])
#                                    return(litePos)})
#       litePos <- as.numeric(unlist(litePos))
#       #print(litePos)
#       simplePos <-numeric()
#       simplePos <-apply(matSimple, 1, function(x){simplePos<-c(simplePos,x[1]:x[2])
#                                    return(simplePos)})
#       simplePos <- as.numeric(unlist(simplePos))
#       # print(simplePos)
#       percCovLiteInSimple <- c(percCovLiteInSimple,(length(which(litePos %in% simplePos))/length(litePos))*100)
#       #print((length(which(litePos %in% simplePos))/length(litePos))*100)
#     } 
#   } 
# }

```


Nevertheless, inconsistencies in the measurements of the ratio of disordered amino acids between different methods do not indicate that the metric could not be used as an estimator of global disorder status in proteins. However, we should certainly  be cautious about comparing values originated from different sources. 
If the percentage of disorder residues correlates well with one or more CDK phosphorylation target properties it could be informative regardless if the estimation is highly accurate.  

```{r composition ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=10}

yeastSeq<-yeastDiso$sequence
names(yeastSeq)<-yeastDiso$acc
yeastSeq<-Biostrings::AAStringSet(yeastSeq)


##DB##

# Since the derived assesment of disorder doen't cover the whole proteome I redifine the universe to those protein that have been evaluated.
yeastSeqDBDiso <- getDBDiso(yeastDiso,"sequences")
yeastSeqDBDiso <- Biostrings::AAStringSet(unlist(yeastSeqDBDiso))
yeastCompDBDiso<-colSums(alphabetFrequency(yeastSeqDBDiso))[1:20]/sum(colSums(alphabetFrequency(yeastSeqDBDiso))[1:20])

yeastSeqDBAll <- yeastSeq[names(yeastSeq) %in% names(yeastSeqDBDiso)] 
yeastCompDBALL<-colSums(alphabetFrequency(yeastSeqDBAll))[1:20]/sum(colSums(alphabetFrequency(yeastSeqDBAll))[1:20])

DBDifferentialComp <- sort((yeastCompDBDiso-yeastCompDBALL)/yeastCompDBALL,decreasing = T)

######################Set the rainbow color palette for the DB set #####################
colorPallete_DBOrdered <- rainbow(20)
names(colorPallete_DBOrdered) <- reorder(names(DBDifferentialComp),-DBDifferentialComp)
########################################################################################

plotDBDifferentialComp <- ggplot()+geom_bar(aes(x=reorder(names(DBDifferentialComp),-DBDifferentialComp), fill=reorder(names(DBDifferentialComp),-DBDifferentialComp), y=DBDifferentialComp),stat="identity") + scale_fill_manual(values=colorPallete_DBOrdered)+ theme_linedraw() + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank()) + scale_y_continuous(breaks = seq(-1,1,0.2),limits = c(-0.9,0.9)) + ggtitle("DB curated")
##Derived##

# Since the derived assesment of disorder doen't cover the whole proteome I redifine the universe to those protein that have been evaluated.
yeastSeqDerivedDiso <- getDerivedDiso(yeastDiso,"sequences")
yeastSeqDerivedDiso <- Biostrings::AAStringSet(unlist(yeastSeqDerivedDiso))
yeastCompDerivedDiso<-colSums(alphabetFrequency(yeastSeqDerivedDiso))[1:20]/sum(colSums(alphabetFrequency(yeastSeqDerivedDiso))[1:20])

yeastSeqDerivedAll <- yeastSeq[names(yeastSeq) %in% names(yeastSeqDerivedDiso)] 
yeastCompDerivedALL<-colSums(alphabetFrequency(yeastSeqDerivedAll))[1:20]/sum(colSums(alphabetFrequency(yeastSeqDerivedAll))[1:20])

derivedDifferentialComp <- sort((yeastCompDerivedDiso-yeastCompDerivedALL)/yeastCompDerivedALL,decreasing = T)

plotDerivedDifferentialComp <- ggplot()+geom_bar(aes(x=reorder(names(derivedDifferentialComp),-derivedDifferentialComp), fill=reorder(names(derivedDifferentialComp),-derivedDifferentialComp), y=derivedDifferentialComp),stat="identity") + scale_fill_manual(values=colorPallete_DBOrdered)+ theme_linedraw() + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank()) + scale_y_continuous(breaks = seq(-1,1,0.2),limits = c(-0.9,0.9)) + ggtitle("Derived disorder")
##SPOT##
yeastCompALL<-colSums(alphabetFrequency(yeastSeq))[1:20]/sum(colSums(alphabetFrequency(yeastSeq))[1:20])

yeastSeqSpotDiso <- getSpotPredDiso(yeastDiso,"sequences","acc")
yeastSeqSpotDiso <- Biostrings::AAStringSet(unlist(yeastSeqSpotDiso))
yeastCompSpotDiso<-colSums(alphabetFrequency(yeastSeqSpotDiso))[1:20]/sum(colSums(alphabetFrequency(yeastSeqSpotDiso))[1:20])
spotDifferentialComp <- sort((yeastCompSpotDiso-yeastCompALL)/yeastCompALL,decreasing = T)

plotSpotDifferentialComp <- ggplot()+geom_bar(aes(x=reorder(names(spotDifferentialComp),-spotDifferentialComp), y=spotDifferentialComp,fill=reorder(names(spotDifferentialComp),-spotDifferentialComp)),stat="identity") + scale_fill_manual(values=colorPallete_DBOrdered)+ theme_linedraw() + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank()) + scale_y_continuous(breaks = seq(-1,1,0.2),limits = c(-0.9,0.9)) + ggtitle("SPOT predictor")

##Simple##

yeastCompALL<-colSums(alphabetFrequency(yeastSeq))[1:20]/sum(colSums(alphabetFrequency(yeastSeq))[1:20])

yeastSeqSimpleDiso <- getPredDiso(yeastDiso,"sequences")
yeastSeqSimpleDiso <- Biostrings::AAStringSet(unlist(yeastSeqSimpleDiso))
yeastCompSimpleDiso<-colSums(alphabetFrequency(yeastSeqSimpleDiso))[1:20]/sum(colSums(alphabetFrequency(yeastSeqSimpleDiso))[1:20])
SimpleDifferentialComp <- sort((yeastCompSimpleDiso-yeastCompALL)/yeastCompALL,decreasing = T)

plotSimpleDifferentialComp <- ggplot()+geom_bar(aes(x=reorder(names(SimpleDifferentialComp),-SimpleDifferentialComp), y=SimpleDifferentialComp,fill=reorder(names(SimpleDifferentialComp),-SimpleDifferentialComp)),stat="identity") + scale_fill_manual(values=colorPallete_DBOrdered)+ theme_linedraw() + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank()) + scale_y_continuous(breaks = seq(-1,1,0.2),limits = c(-0.9,0.9)) + ggtitle("Simple predictor")
##Lite##

yeastSeqLiteDiso <- getPredLiteDiso(yeastDiso,"sequences")
yeastSeqLiteAll <- yeastSeq[names(yeastSeq) %in% names(yeastSeqLiteDiso)] 
yeastCompLiteALL<-colSums(alphabetFrequency(yeastSeqLiteAll))[1:20]/sum(colSums(alphabetFrequency(yeastSeqLiteAll))[1:20])

yeastSeqLiteDiso <- Biostrings::AAStringSet(unlist(yeastSeqLiteDiso))
yeastCompLiteDiso<-colSums(alphabetFrequency(yeastSeqLiteDiso))[1:20]/sum(colSums(alphabetFrequency(yeastSeqLiteDiso))[1:20])
LiteDifferentialComp <- sort((yeastCompLiteDiso-yeastCompLiteALL)/yeastCompLiteALL,decreasing = T)

plotLiteDifferentialComp <- ggplot() + geom_bar(aes(x=reorder(names(LiteDifferentialComp),-LiteDifferentialComp), y=LiteDifferentialComp,fill=reorder(names(LiteDifferentialComp),-LiteDifferentialComp)),stat="identity") + scale_fill_manual(values=colorPallete_DBOrdered)+ theme_linedraw() + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank()) + scale_y_continuous(breaks = seq(-1,1,0.2),limits = c(-0.9,0.9)) + ggtitle("Lite predictor")

fig<-ggarrange(plotDBDifferentialComp,plotDerivedDifferentialComp,plotSimpleDifferentialComp,plotLiteDifferentialComp,plotSpotDifferentialComp,ncol = 1)
annotate_figure(fig,left = text_grob("Comp. Disorder - Comp. Universe / Comp. Universe", rot = 90,size = 16))
```


# Cdk1 targets  
  

  
## Ubersax 2003  
  

For defining the targets for Cdk1 in Saccharomyces cerevisiae, I used publicly available data published by David Morgan's group [@Ubersax2003]. In this paper, the authors used a modified version of Cdk1 that allows bulky ATP analogues to bind in the ATP binding pocket (Cdk1-as1). They have used a library of yeast strains, each expressing a unique ORF fused with GST.  In this way, a radiolabelled bulky ATP analogue, N6-(benzyl) ATP, is administered to _in vitro_ perform kinase reactions between purified Cdk1-as1·Clb2 and the lysates from 695 different strains from the library. From the 695 unique ORF-GST selected,  385 correspond to ORFs encodign proteins with two or more Cdk consensus phosphorylation sites (S/T*-P-x-K/R), 137 ORFs encoding proteins with a single Cdk consensus site and whose transcripts are cell-cycle regulated, and 183 randomly chosen ORFs.
For each reaction, they performed a step of GST tag mediated purification and they measured the levels of tagged phosphate. The authors define the measure __"P-Score"__ as the logarithm of the amount of phosphate incorporated per nanogram of protein, and they have set the treshold of P-Score > 2 for designate the Cdk1 targets.



```{r cdk1 targets summary,echo=FALSE,message=FALSE,warning=FALSE}

# _____________________________________________________________



# npredictorsCDK=0
# nmobidbliteCDK=0
# nsimpleCDK=0
# #methods: mobidb-lite, simple
# for (i in 1:nrow(yeastDisoCDK1target)) {
#   if(!is.null(yeastDisoCDK1target[i,5]$predictors[[1]])){
#     npredictorsCDK=npredictorsCDK+1
#     }
#   if( "mobidb-lite" %in% yeastDisoCDK1target[i,5]$predictors[[1]]$method){
#     nmobidbliteCDK=nmobidbliteCDK+1
#     }
#   if( "simple" %in% yeastDisoCDK1target[i,5]$predictors[[1]]$method){
#     nsimpleCDK=nsimpleCDK+1
#     }  
#   }
# 
# 
# #methods: full bfactor missing_residues mobile
# nderivedCDK=0
# nfullCDK=0
# nbfactorCDK=0
# nmissing_residuesCDK=0
# nmobileCDK=0
# 
# for (i in 1:nrow(yeastDisoCDK1target)) {
#   if(!is.null(yeastDisoCDK1target[i,5]$derived[[1]])){
#     nderivedCDK=nderivedCDK+1
#   }
#   if( "full" %in% yeastDisoCDK1target[i,5]$derived[[1]]$method){
#     nfullCDK=nfullCDK+1
#   }
#   if( "bfactor" %in% yeastDisoCDK1target[i,5]$derived[[1]]$method){
#     nbfactorCDK=nbfactorCDK+1
#   }
#   if( "missing_residues" %in% yeastDisoCDK1target[i,5]$derived[[1]]$method){
#     nmissing_residuesCDK=nmissing_residuesCDK+1
#   }
#   if( "mobile" %in% yeastDisoCDK1target[i,5]$derived[[1]]$method){
#     nmobileCDK=nmobileCDK+1
#   }
# }
# #methods: full
# ndbCDK=0
# for (i in 1:nrow(yeastDisoCDK1target)) {
#   if(!is.null(yeastDisoCDK1target[i,5]$db[[1]])){
#     ndbCDK=ndbCDK+1
#     }
# }
```

```{r cdkTargetTable ,echo=FALSE,message=FALSE,warning=FALSE,rows.print=20}
datatable(Ubersax_data[order(Ubersax_data$`P-Score`,decreasing = T),c(1,11,15,6,7,8,3,4,5)],options = list(sDom  = '<"top">lrt<"bottom">ip'))

write.table(Ubersax_data[order(Ubersax_data$`P-Score`,decreasing = T),c(1,11,15,6,7,8,3,4,5)],"Ubersax_data_download.tab",quote = F,sep = "\t",row.names = F)
```
  
From the original 695 ORFs, I kept 687 which could be unequivocally mapped to a unique Uniprot accesion. This ID will allow us to merge the disorder data obtained from the MobiDB database and the annotations extracted from the Cdk1 targets paper data. The final data is available [here](Ubersax_data_download.tab)  
  
### P-Score
  
In order to understand the P-Score, I compare its value against P-score normalized by the numbers of mol proteins instead of the mass. For this calculation we consider that the average molar mass of one amino acid is 120 gr/mol:

$$\log(\frac{Phosphorylation_{<arbitrary\ units>}}{Protein Level _{<ng>} / MW _{<ng/nmol>}}) = log(\frac{Phosphorylation_{<arbitrary\ units>}}{Protein Level _{<ng>} / (length _{<amin oacids>} *\ 120 _{<ng/(nmol*amino acids)>}))})$$   
  
If we consider every consensus phosphosite as phosphorylatable, we could normalize not only by the number of moles in the reaction, but also by the number of moles of phosphosites. We could think this concept as similar to the definition of one Equivalent. For the two consensus patterns considered (S/T\*-P-x-K/R and S/T\*-P), we could calculate the the number of equivalents of a phsophosite as: 
$$nEq/mol*nmol = n° (S/T-P-x-K/R)\ or\ n°(S/T-P)*\frac{Protein Level _{<ng>}}{length _{<amin oacids>} *\ 120 _{<ng/(nmol*amino acids)>}}$$
For understand if the P-Score is the best metric for asses the levels of CDK activity over an specific target, I compared the levels of Phosphorylation versus the Protein Levels (in nanograms) and the normalization of the protein levels considering the number of nano moles and the numbers of nano equivalents of phosho sites, instead of using only the mass.  
  


In the logarithmic scale the difference of considering only the mass, or the number of nano moles/nano equivalents does not show striking effects on the distribution of the dots in the [supplementary figure 6](../supplementary/Supp6.html). Then, direct comparisons between the mass of protein meassured and the calculation of moles and equivalents were performed. Is possible to see some dispersion of the dots for high values of protein levels, but still a direct linear correlation between the grmas of protein meassured and the calculations for moles and equivalents. In logarithmic scale there is even a stronger dependency.
All in all this means that the posible cofunding variables in the protein level meassureements that I have used for normalization (molecular weight and phosphosites) are, in fact, constants.  


```{r cdkTarget ,echo=FALSE,message=FALSE,warning=FALSE, fig.cap="Figure 6:",fig.height=10}

scoresData <- Ubersax_data[,c("Phosphorylation","P-Score","Protein Level","nmol","nEq_STP","nEq_STPXKR","target","[ST]-P","[ST]-P-X-[KR]")]
scoresData$logPhospho <- log10(Ubersax_data$Phosphorylation)
scoresData$logMol <- log10(Ubersax_data$nmol)
scoresData$logEq_STP <- log10(Ubersax_data$nEq_STP)
scoresData$logEq_STPXR <- log10(Ubersax_data$nEq_STPXKR)
scoresData$targetStrict <- cut(Ubersax_data$`P-Score`,breaks = c(-Inf,3,Inf),labels = c("Non Cdk1 target","Cdk1 target"),right = F)

l_pl_nm<-ggplot(scoresData,aes(x=log10(`Protein Level`),y=log10(`nmol`)))+geom_point(aes(color=target)) + theme_linedraw() + theme(axis.title.x = element_blank()) +  xlab("Log10(Protein Level <ng>)") + ylab("Log10(nMoles)")

l_pl_nES<-ggplot(scoresData,aes(x=log10(`Protein Level`),y=log10(`nEq_STP`)))+geom_point(aes(color=target)) + theme_linedraw() + theme(axis.title.x = element_blank()) +  xlab("Log10(Protein Level <ng>)") + ylab("Log10(nEquivalents S/T-P)")
l_pl_nEF<-ggplot(scoresData,aes(x=log10(`Protein Level`),y=log10(`nEq_STPXKR`)))+geom_point(aes(color=target)) + theme_linedraw() + xlab("Log10(Protein Level <ng>)") + ylab("Log10(nEquivalents S/T-P-x-K/R)")




pl_nm<-ggplot(scoresData,aes(x=`Protein Level`,y=`nmol`))+geom_point(aes(color=target)) + theme_linedraw()+ theme(axis.title.x = element_blank())  +  xlab("Protein Level <ng>") + ylab("nMoles")

pl_nES<-ggplot(scoresData,aes(x=`Protein Level`,y=`nEq_STP`))+geom_point(aes(color=target)) + theme_linedraw() + theme(axis.title.x =  element_blank()) +  xlab("Protein Level <ng>") + ylab("nEquivalents S/T-P")
pl_nEF<-ggplot(scoresData,aes(x=`Protein Level`,y=`nEq_STPXKR`))+geom_point(aes(color=target)) + theme_linedraw() + xlab("Protein Level <ng>") + ylab("nEquivalents S/T-P-x-K/R")

ggarrange(l_pl_nm,pl_nm,l_pl_nES,pl_nES,l_pl_nEF,pl_nEF,nrow = 3, ncol = 2,common.legend = TRUE, legend = "bottom")

```
In order to confirm the lack of response of the P-Score on the different normalizations of the protein levels, the values of this metric were compared against the values of the re-calculated P-Scores considering the number of nano moles or nano equivalents of phosphorylatable protein. In all 3 cases the difference between the P-Scores is not significant. As an example, if we consider a cutoff of 6 for the logarithm of the phosphorylation divided by the nano equivalents of [ST]P phosphosites, only 18 values are not consistent between both of the predictions.
In practical terms, the P-Score doesn't need to be normalized by the length of the protein nor the density of phosphosites.

P-score does not consider the size nor the number of consensus phospho sites for Cdk1. P-Score = log(Phosphorylation<arbitrary units>/ Protein level <ng>)

For correcting for protein lenght we can do P-Score Mol = log(Phosphorylation<arbitrary units>/ number of protein umols) = log(Phosphorylation<arbitrary units>/ (Protein Level <ng> / (length <AAs>)*120 <g mol/AAs>))

Correcting for number of moles in the reaction or for number of Eq of [ST]-P sites or [ST]-P-X-[KR] doesn't change much, probably due to the fact that this correction is linear and the final effect in the logarithm scale of the Score is not relevant  
  
## Holt 2009
  
[@Holt2009]  
  

# Protein disorder in Cdk1 targets

## Distribution of disorder percentage for Cdk1 tagets in yeast {.tabset .tabset-fade}

### Ubersax
```{r Ubersax disorder percentage,echo=FALSE,message=FALSE,warning=FALSE}


dens_simpleDisoPerc<-ggplot(Ubersax_data)+geom_density(aes(x = predDisoPerc,fill = target),alpha=.5,bw=5)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5)) + scale_y_continuous(limits = c(0,0.05)) + scale_x_continuous(name="% of predicted Disorder",limits = c(0,100))+ggtitle("Simple Predictor")
dens_liteDisoPerc<-ggplot(subset(Ubersax_data,!is.na(predLiteDisoPerc)))+geom_density(aes(x = predLiteDisoPerc,fill = target),alpha=.5,bw=5)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5)) + theme(axis.title.y = element_blank())+ scale_y_continuous(limits = c(0,0.05)) + scale_x_continuous(name="% of predicted Disorder",limits = c(0,100)) + theme(axis.title.y = element_blank())+labs(title="MobiDB-Lite Predictor")
dens_spotDisoPerc<-ggplot(Ubersax_data)+geom_density(aes(x = predSpotDisoPerc,fill = target),alpha=.5,bw=5)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5)) + theme(axis.title.y = element_blank())+ scale_y_continuous(limits = c(0,0.05)) + scale_x_continuous(name="% of predicted Disorder",limits = c(0,100)) + theme(axis.title.y = element_blank())+labs(title="SPOT Predictor")

hist_simpleDisoPerc<-ggplot(Ubersax_data)+geom_histogram(aes(x = predDisoPerc,fill = target),alpha=.5,bins = 50)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5))  + scale_x_continuous(name="% of predicted Disorder",limits = c(-2,102))
hist_liteDisoPerc<-ggplot(subset(Ubersax_data,!is.na(predLiteDisoPerc)))+geom_histogram(aes(x = predLiteDisoPerc,fill = target),alpha=.5,bins = 50)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5))  + theme(axis.title.y = element_blank())+  scale_x_continuous(name="% of predicted Disorder",limits = c(-2,102)) + theme(axis.title.y = element_blank())
hist_spotDisoPerc<-ggplot(Ubersax_data)+geom_histogram(aes(x = predSpotDisoPerc,fill = target),alpha=.5,bins = 50)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5)) + theme(axis.title.y = element_blank())+ scale_x_continuous(name="% of predicted Disorder",limits = c(-2,102)) + theme(axis.title.y = element_blank())

ggarrange(dens_simpleDisoPerc,dens_liteDisoPerc,dens_spotDisoPerc,hist_simpleDisoPerc,hist_liteDisoPerc,hist_spotDisoPerc,nrow = 2, ncol = 3,common.legend = TRUE,legend = "bottom", align = "hv")

#Var and SD cheking. t-tests
ubersax_simple_NonCdk1_VAR <- var(Ubersax_data$predDisoPerc[Ubersax_data$target=="Non Cdk1 target"],na.rm = T)
ubersax_simple_Cdk1_VAR <- var(Ubersax_data$predDisoPerc[Ubersax_data$target=="Cdk1 target"],na.rm = T)
ubersax_simple_Cdk1_SD <- sd(Ubersax_data$predDisoPerc[Ubersax_data$target=="Cdk1 target"],na.rm = T)
ubersax_simple_NonCdk1_SD <- sd(Ubersax_data$predDisoPerc[Ubersax_data$target=="Non Cdk1 target"],na.rm = T)
ubersax_simple_test <- t.test(Ubersax_data$predDisoPerc[Ubersax_data$target=="Cdk1 target"],Ubersax_data$predSpotDisoPerc[Ubersax_data$target=="Non Cdk1 target"],var.equal=T)

ubersax_lite_NonCdk1_VAR <- var(Ubersax_data$predLiteDisoPerc[Ubersax_data$target=="Non Cdk1 target"],na.rm = T)
ubersax_lite_Cdk1_VAR <- var(Ubersax_data$predLiteDisoPerc[Ubersax_data$target=="Cdk1 target"],na.rm = T)
ubersax_lite_Cdk1_SD <- sd(Ubersax_data$predLiteDisoPerc[Ubersax_data$target=="Cdk1 target"],na.rm = T)
ubersax_lite_NonCdk1_SD <- sd(Ubersax_data$predLiteDisoPerc[Ubersax_data$target=="Non Cdk1 target"],na.rm = T)
ubersax_lite_test <- t.test(Ubersax_data$predLiteDisoPerc[Ubersax_data$target=="Cdk1 target"],Ubersax_data$predSpotDisoPerc[Ubersax_data$target=="Non Cdk1 target"],var.equal=T)

ubersax_spot_NonCdk1_VAR <- var(Ubersax_data$predSpotDisoPerc[Ubersax_data$target=="Non Cdk1 target"])
ubersax_spot_Cdk1_VAR <- var(Ubersax_data$predSpotDisoPerc[Ubersax_data$target=="Cdk1 target"],na.rm = T)
ubersax_spot_Cdk1_SD <- sd(Ubersax_data$predSpotDisoPerc[Ubersax_data$target=="Cdk1 target"],na.rm = T)
ubersax_spot_NonCdk1_SD <- sd(Ubersax_data$predSpotDisoPerc[Ubersax_data$target=="Non Cdk1 target"],na.rm = T)
ubersax_spot_test <- t.test(Ubersax_data$predSpotDisoPerc[Ubersax_data$target=="Cdk1 target"],Ubersax_data$predSpotDisoPerc[Ubersax_data$target=="Non Cdk1 target"],var.equal=T)

# col1 <- rbind(ubersax_simple_NonCdk1_VAR,ubersax_simple_NonCdk1_SD,ubersax_simple_test$estimate[2],ubersax_simple_test$p.value)
# col2 <- rbind(ubersax_simple_Cdk1_VAR,ubersax_simple_Cdk1_SD,ubersax_simple_test$estimate[1],ubersax_simple_test$p.value)
# col3 <- rbind(ubersax_lite_NonCdk1_VAR,ubersax_lite_NonCdk1_SD,ubersax_lite_test$estimate[2],ubersax_lite_test$p.value)
# col4 <- rbind(ubersax_lite_Cdk1_VAR,ubersax_lite_Cdk1_SD,ubersax_lite_test$estimate[1],ubersax_lite_test$p.value)
# col5 <- rbind(ubersax_lite_NonCdk1_VAR,ubersax_lite_NonCdk1_SD,ubersax_spot_test$estimate[2],ubersax_lite_test$p.value)
# col6 <- rbind(ubersax_lite_Cdk1_VAR,ubersax_lite_Cdk1_SD,ubersax_spot_test$estimate[1],ubersax_lite_test$p.value)
# ubersaxPercTable <- as.data.frame(cbind(col1,col2,col3,col4,col5,col6))
# rownames(ubersaxPercTable) <- c("Variance","Standar Deviation","Estimated mean","T test p-value")
# colnames(ubersaxPercTable) <-c("col1","col2","col3","col4","col5","col6")
# ubersaxPercTable <- flextable(ubersaxPercTable)
# flextable::add_header(ubersaxPercTable,top = T,c("Non Cdk1 target","Cdk1 target","Non Cdk1 target","Cdk1 target","Non Cdk1 target","Cdk1 target"))
# flextable::add_header(ubersaxPercTable,top = T,c("Simple","Simple","Lite","Lite","Spot","Spot"))
# flextable:::merge_h(ubersaxPercTable,part = "header")
```
  

### Holt

```{r Holt disorder percentage,echo=FALSE,message=FALSE,warning=FALSE}


dens_simpleDisoPerc<-ggplot(Holt_data)+geom_density(aes(x = predDisoPerc,fill = target_consensus),alpha=.5,bw=5)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5)) + scale_y_continuous(limits = c(0,0.05)) + scale_x_continuous(name="% of predicted Disorder",limits = c(0,100))+ggtitle("Simple Predictor")
dens_liteDisoPerc<-ggplot(subset(Holt_data,!is.na(predLiteDisoPerc)))+geom_density(aes(x = predLiteDisoPerc,fill = target_consensus),alpha=.5,bw=5)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5)) + theme(axis.title.y = element_blank())+ scale_y_continuous(limits = c(0,0.05)) + scale_x_continuous(name="% of predicted Disorder",limits = c(0,100)) + theme(axis.title.y = element_blank())+labs(title="MobiDB-Lite Predictor")
dens_spotDisoPerc<-ggplot(Holt_data)+geom_density(aes(x = predSpotDisoPerc,fill = target_consensus),alpha=.5,bw=5)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5)) + theme(axis.title.y = element_blank())+ scale_y_continuous(limits = c(0,0.05)) + scale_x_continuous(name="% of predicted Disorder",limits = c(0,100)) + theme(axis.title.y = element_blank())+labs(title="SPOT Predictor")

hist_simpleDisoPerc<-ggplot(Holt_data)+geom_histogram(aes(x = predDisoPerc,fill = target_consensus),alpha=.5,bins = 50)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5))  + scale_x_continuous(name="% of predicted Disorder",limits = c(-2,102))
hist_liteDisoPerc<-ggplot(subset(Holt_data,!is.na(predLiteDisoPerc)))+geom_histogram(aes(x = predLiteDisoPerc,fill = target_consensus),alpha=.5,bins = 50)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5))  + theme(axis.title.y = element_blank())+  scale_x_continuous(name="% of predicted Disorder",limits = c(-2,102)) + theme(axis.title.y = element_blank())
hist_spotDisoPerc<-ggplot(Holt_data)+geom_histogram(aes(x = predSpotDisoPerc,fill = target_consensus),alpha=.5,bins = 50)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5)) + theme(axis.title.y = element_blank())+ scale_x_continuous(name="% of predicted Disorder",limits = c(-2,102)) + theme(axis.title.y = element_blank())

ggarrange(dens_simpleDisoPerc,dens_liteDisoPerc,dens_spotDisoPerc,hist_simpleDisoPerc,hist_liteDisoPerc,hist_spotDisoPerc,nrow = 2, ncol = 3,common.legend = TRUE,legend = "bottom", align = "hv")

holt_simple_NonCdk1_VAR <- var(Holt_data$predDisoPerc[Holt_data$target_consensus=="Non Cdk1 target"],na.rm = T)
holt_simple_Cdk1_VAR <- var(Holt_data$predDisoPerc[Holt_data$target_consensus=="Cdk1 target"],na.rm = T)
holt_simple_Cdk1_SD <- sd(Holt_data$predDisoPerc[Holt_data$target_consensus=="Cdk1 target"],na.rm = T)
holt_simple_NonCdk1_SD <- sd(Holt_data$predDisoPerc[Holt_data$target_consensus=="Non Cdk1 target"],na.rm = T)
holt_simple_test <- t.test(Holt_data$predDisoPerc[Holt_data$target_consensus=="Cdk1 target"],Holt_data$predSpotDisoPerc[Holt_data$target_consensus=="Non Cdk1 target"],var.equal=T)

holt_lite_NonCdk1_VAR <- var(Holt_data$predLiteDisoPerc[Holt_data$target_consensus=="Non Cdk1 target"],na.rm = T)
holt_lite_Cdk1_VAR <- var(Holt_data$predLiteDisoPerc[Holt_data$target_consensus=="Cdk1 target"],na.rm = T)
holt_lite_Cdk1_SD <- sd(Holt_data$predLiteDisoPerc[Holt_data$target_consensus=="Cdk1 target"],na.rm = T)
holt_lite_NonCdk1_SD <- sd(Holt_data$predLiteDisoPerc[Holt_data$target_consensus=="Non Cdk1 target"],na.rm = T)
holt_lite_test <- t.test(Holt_data$predLiteDisoPerc[Holt_data$target_consensus=="Cdk1 target"],Holt_data$predSpotDisoPerc[Holt_data$target_consensus=="Non Cdk1 target"],var.equal=T)

holt_spot_NonCdk1_VAR <- var(Holt_data$predSpotDisoPerc[Holt_data$target_consensus=="Non Cdk1 target"])
holt_spot_Cdk1_VAR <- var(Holt_data$predSpotDisoPerc[Holt_data$target_consensus=="Cdk1 target"],na.rm = T)
holt_spot_Cdk1_SD <- sd(Holt_data$predSpotDisoPerc[Holt_data$target_consensus=="Cdk1 target"],na.rm = T)
holt_spot_NonCdk1_SD <- sd(Holt_data$predSpotDisoPerc[Holt_data$target_consensus=="Non Cdk1 target"],na.rm = T)
holt_spot_test <- t.test(Holt_data$predSpotDisoPerc[Holt_data$target_consensus=="Cdk1 target"],Holt_data$predSpotDisoPerc[Holt_data$target_consensus=="Non Cdk1 target"],var.equal=T)
```
  
For boths datasets there is a statistical significative increase in the percentage of predicted disorder aminoacids in yeast proteins for those proteins characterized as CDK1 targets. Moreover, for the Holt dataset a similar result is obtained if we skipped the CDK1 phosporylation site filter. That is, CDK1 targets defined as all the proteins with Serines and Threonines residues presenting a decrease on the phosphorylation upon CDK1 inhibition,regardles if they respect the minimal CDK1 phosphosite consensus motif, show an increase in the percentage of predicted disorder amino acids (<a href="supplementary/Supp.html">Supplementary</a> Fig X).



```{r Ubersax agreements disorder vs phosposites ,echo=FALSE,message=FALSE,warning=FALSE}



####Lite####
UbxAllLiteDisorderedRegions <- getPredLiteDiso(df = yeastDiso_Ubersax_subset,"indices")
UbxAllLiteStratContingencyArray <- getStratContingencyArray(df = yeastDiso_Ubersax_subset,sequence_col = "sequence",disordered_index_list = UbxAllLiteDisorderedRegions)

woolf_test(UbxAllLiteStratContingencyArray)
fisherLiteDF <- data.frame(Ubersax_data$Entry)
colnames(fisherLiteDF) <- "acc"
fisherLiteDF$target <- Ubersax_data$target
fisherLiteDF$ORs <- apply(UbxAllLiteStratContingencyArray, 3,function(x){
     ft<-fisher.test(x)
     return(ft$estimate)
     })
fisherLiteDF$pvalue <- apply(UbxAllLiteStratContingencyArray, 3,function(x){
     ft<-fisher.test(x)
     return(ft$p.value)
     })
fisherLiteDF$adjPvalue <- p.adjust(fisherLiteDF$pvalue,method = "BH")
fisherLiteDF$logAdjPvalue <- -log10(fisherLiteDF$adjPvalue)
fisherLiteDF$logOr <- log10(fisherLiteDF$ORs)

fisherLiteDF$logOr[is.infinite(fisherLiteDF$logOr) & fisherLiteDF$logOr > 0] <- 2
fisherLiteDF$logOr[is.infinite(fisherLiteDF$logOr) & fisherLiteDF$logOr < 0] <- -2


###Trying to find a way to move the dots out from the logPvalue=0 with random noise(Jitter) in order to asses the amount of points we have there.
set.seed(1234)
fisherLiteDF$logAdjPvalue[fisherLiteDF$logAdjPvalue==0]<- 0.1
fisherLiteDF$logAdjPvalue[fisherLiteDF$logAdjPvalue==0.1]<-abs(jitter(fisherLiteDF$logAdjPvalue[fisherLiteDF$logAdjPvalue==0.1],6))

ggplot(fisherLiteDF, aes(x = logOr, y = logAdjPvalue)) +
  geom_point(aes(color = target),alpha=0.6) +
  # scale_color_manual(values = c("", "grey")) +
  theme_bw(base_size = 12) + theme(legend.position = "bottom") +
  scale_y_continuous(trans='log10',limits = c(0.1,10)) +
  xlim(-2, 2) +
  scale_shape_manual()+
  # geom_point(data = subset(fisherLiteDF, log10(ORs) < -2), aes(x = -2, y = -log10(pvalue)), alpha = 0.5,shape=60,size=4) +
  # geom_point(data = subset(fisherLiteDF, log10(ORs) > 2), aes(x = 2, y = -log10(pvalue)), alpha = 0.5,shape=62,size=4) +
  geom_hline(yintercept=-log10(0.05), linetype="dashed", color = "red")+
  annotate(geom = "text",label="FDR = 0.05",x=-2,y=-log10(0.05),hjust=0,vjust=-0.8)

####SPOT####
UbxAllSPOTDisorderedRegions <- getSpotPredDiso(df = yeastDiso_Ubersax_subset,"indices",accession_col = "acc")
UbxAllSPOTStratContingencyArray <- getStratContingencyArray(df = yeastDiso_Ubersax_subset,sequence_col = "sequence",disordered_index_list = UbxAllSPOTDisorderedRegions)

woolf_test(UbxAllSPOTStratContingencyArray)
fisherSpotDF <- data.frame(Ubersax_data$Entry)
colnames(fisherSpotDF) <- "acc"
fisherSpotDF$target <- Ubersax_data$target
fisherSpotDF$ORs <- apply(UbxAllSPOTStratContingencyArray, 3,function(x){
     ft<-fisher.test(x)
     return(ft$estimate)
     })
fisherSpotDF$pvalue <- apply(UbxAllSPOTStratContingencyArray, 3,function(x){
     ft<-fisher.test(x)
     return(ft$p.value)
     })
fisherSpotDF$adjPvalue <- p.adjust(fisherSpotDF$pvalue,method = "BH")
fisherSpotDF$logAdjPvalue <- -log10(fisherSpotDF$adjPvalue)
fisherSpotDF$logOr <- log10(fisherSpotDF$ORs)

fisherSpotDF$logOr[is.infinite(fisherSpotDF$logOr) & fisherSpotDF$logOr > 0] <- 2
fisherSpotDF$logOr[is.infinite(fisherSpotDF$logOr) & fisherSpotDF$logOr < 0] <- -2


###Trying to find a way to move the dots out from the logPvalue=0 with random noise(Jitter) in order to asses the amount of points we have there.
set.seed(1234)
fisherSpotDF$logAdjPvalue[fisherSpotDF$logAdjPvalue==0]<- 0.1
fisherSpotDF$logAdjPvalue[fisherSpotDF$logAdjPvalue==0.1]<-abs(jitter(fisherSpotDF$logAdjPvalue[fisherSpotDF$logAdjPvalue==0.1],6))

ggplot(fisherSpotDF, aes(x = logOr, y = logAdjPvalue)) +
  geom_point(aes(color = target),alpha=0.6) +
  # scale_color_manual(values = c("", "grey")) +
  theme_bw(base_size = 12) + theme(legend.position = "bottom") +
  scale_y_continuous(trans='log10',limits = c(0.1,10)) +
  xlim(-2, 2) +
  scale_shape_manual()+
  # geom_point(data = subset(fisherDF, log10(ORs) < -2), aes(x = -2, y = -log10(pvalue)), alpha = 0.5,shape=60,size=4) +
  # geom_point(data = subset(fisherDF, log10(ORs) > 2), aes(x = 2, y = -log10(pvalue)), alpha = 0.5,shape=62,size=4) +
  geom_hline(yintercept=-log10(0.05), linetype="dashed", color = "red")+
  annotate(geom = "text",label="FDR = 0.05",x=-2,y=-log10(0.05),hjust=0,vjust=-0.8)


# #####################
# liteDisorderedRegions <- getPredLiteDiso(df = yeastDisoCDK1target[which(Ubersax_data$`P-Score`>=2 &!is.na(Ubersax_data$predLiteDisoPercCdkTargets)),],"indices")
# liteTPphosphoSites <- gregexpr("[ST]P",yeastDisoCDK1target[which(Ubersax_data$`P-Score`>=2 &!is.na(Ubersax_data$predLiteDisoPercCdkTargets)),]$sequence)
# 
# liteStratContingencyArray <- array(dim = c(2,2,nrow(yeastDisoCDK1target[which(Ubersax_data$`P-Score`>=2 &!is.na(Ubersax_data$predLiteDisoPercCdkTargets)),])))
# for (i in 1:nrow(yeastDisoCDK1target[which(Ubersax_data$`P-Score`>=2 &!is.na(Ubersax_data$predLiteDisoPercCdkTargets)),])) {
# 
#   indexProtLite <- 1:nchar(yeastDisoCDK1target[which(Ubersax_data$`P-Score`>=2 &!is.na(Ubersax_data$predLiteDisoPercCdkTargets)),]$sequence[[i]])
#   indexDisoLite <- liteDisorderedRegions[[i]]
#   indexOrdLite <- setdiff(indexProtLite,indexDisoLite)
#   indexPhosphoLite <- as.numeric(liteTPphosphoSites[[i]])
#   indexNonPhosphoLite <- setdiff(indexProtLite,indexPhosphoLite)
# 
#   ndpLite <- length(which(indexPhosphoLite %in% indexDisoLite))
#   nopLite <- length(which(indexPhosphoLite %in% indexOrdLite))
#   ndnpLite <- length(which(indexNonPhosphoLite %in% indexDisoLite))
#   nonpLite<- length(which(indexNonPhosphoLite %in% indexOrdLite))
#   liteStratContingencyArray[1,1,i]<-ndpLite
#   liteStratContingencyArray[1,2,i]<-nopLite
#   liteStratContingencyArray[2,1,i]<-ndnpLite
#   liteStratContingencyArray[2,2,i]<-nonpLite
# 
# }
# 
# woolf_test(liteStratContingencyArray)
# ors <- apply(liteStratContingencyArray, 3,function(x){
#      ft<-fisher.test(x)
#      return(ft$estimate)
#      })
# 
# logors <- log10(ors)
# logors[is.infinite(logors) & logors > 0] <- 2
# logors[is.infinite(logors) & logors < 0] <- -2
# 
# ggplot()+geom_bar(aes(x=1:length(ors), y=ors),stat="identity")+ylim(c(-2,2))
 


####Ubersax####

# disorderedRegions <- getPredDiso(df = yeastDiso,"indices")
# TPphosphoSites <- gregexpr("[ST]P",yeastDiso$sequence)
# 
# stratContingencyArray <- array(dim = c(2,2,nrow(yeastDiso)))
# for (i in 1:nrow(yeastDiso)) {
#   
#   indexProt <- 1:nchar(yeastDiso$sequence[[i]])
#   indexDiso <- disorderedRegions[[i]]
#   indexOrd <- setdiff(indexProt,indexDiso)
#   indexPhospho <- as.numeric(TPphosphoSites[[i]])
#   indexNonPhospho <- setdiff(indexProt,indexPhospho)
#   
#   ndp <- length(which(indexPhospho %in% indexDiso))
#   nop <- length(which(indexPhospho %in% indexOrd))
#   ndnp <- length(which(indexNonPhospho %in% indexDiso))
#   nonp<- length(which(indexNonPhospho %in% indexOrd))
#   stratContingencyArray[1,1,i]<-ndp
#   stratContingencyArray[1,2,i]<-nop
#   stratContingencyArray[2,1,i]<-ndnp
#   stratContingencyArray[2,2,i]<-nonp
#   
# }
# woolf_test(stratContingencyArray)
# apply(stratContingencyArray, 3,function(x){
#      ft<-fisher.test(x)
#      return(ft$p.value)
#      })
# 
# 
# 
# 
# liteDisorderedRegions <- getPredLiteDiso(df = yeastDiso,"indices")
# liteTPphosphoSites <- gregexpr("[ST]P",yeastDiso$sequence)
# 
# liteStratContingencyArray <- array(dim = c(2,2,nrow(yeastDiso)))
# for (i in 1:nrow(yeastDiso)) {
#   
#   indexProtLite <- 1:nchar(yeastDiso$sequence[[i]])
#   indexDisoLite <- liteDisorderedRegions[[i]]
#   indexOrdLite <- setdiff(indexProtLite,indexDisoLite)
#   indexPhosphoLite <- as.numeric(liteTPphosphoSites[[i]])
#   indexNonPhosphoLite <- setdiff(indexProtLite,indexPhosphoLite)
#   
#   ndpLite <- length(which(indexPhosphoLite %in% indexDisoLite))
#   nopLite <- length(which(indexPhosphoLite %in% indexOrdLite))
#   ndnpLite <- length(which(indexNonPhosphoLite %in% indexDisoLite))
#   nonpLite<- length(which(indexNonPhosphoLite %in% indexOrdLite))
#   liteStratContingencyArray[1,1,i]<-ndpLite
#   liteStratContingencyArray[1,2,i]<-nopLite
#   liteStratContingencyArray[2,1,i]<-ndnpLite
#   liteStratContingencyArray[2,2,i]<-nonpLite
#   
# }

```


```{r Holt agreements disorder vs phosposites ,echo=FALSE,message=FALSE,warning=FALSE}
####Lite####
HltAllLiteDisorderedRegions <- getPredLiteDiso(df = yeastDiso_Ubersax_subset,"indices")
HltAllLiteStratContingencyArray <- getStratContingencyArray(df = yeastDiso_Ubersax_subset,sequence_col = "sequence",disordered_index_list = HltAllLiteDisorderedRegions)

woolf_test(HltAllLiteStratContingencyArray)
Hlt_fisherDF <- data.frame(Holt_data$Entry)
colnames(fisherDF) <- "acc"
Hlt_fisherDF$target <- Ubersax_data$target
Hlt_fisherDF$ORs <- apply(HltAllLiteStratContingencyArray, 3,function(x){
     ft<-fisher.test(x)
     return(ft$estimate)
     })
Hlt_fisherDF$pvalue <- apply(HltAllLiteStratContingencyArray, 3,function(x){
     ft<-fisher.test(x)
     return(ft$p.value)
     })
Hlt_fisherDF$logPvalue <- -log10(Hlt_fisherDF$pvalue)
Hlt_fisherDF$logOr <- log10(Hlt_fisherDF$ORs)

Hlt_fisherDF$logOr[is.infinite(Hlt_fisherDF$logOr) & Hlt_fisherDF$logOr > 0] <- 2
Hlt_fisherDF$logOr[is.infinite(Hlt_fisherDF$logOr) & Hlt_fisherDF$logOr < 0] <- -2


###Trying to find a way to move the dots out from the logPvalue=0 with random noise(Jitter) in order to asses the amount of points we have there.
# set.seed(1234)
# fisherDF$pvalue[fisherDF$pvalue==1]<-abs(jitter(fisherDF$logPvalue[fisherDF$logPvalue==0]))

ggplot(Hlt_fisherDF, aes(x = logOr, y = logPvalue)) +
  geom_point(aes(color = target)) +
  # scale_color_manual(values = c("", "grey")) +
  theme_bw(base_size = 12) + theme(legend.position = "bottom") +
  scale_y_continuous(trans='log10',limits = c(0.1,10)) +
  xlim(-2, 2) +
  scale_shape_manual()+
  # geom_point(data = subset(fisherDF, log10(ORs) < -2), aes(x = -2, y = -log10(pvalue)), alpha = 0.5,shape=60,size=4) +
  # geom_point(data = subset(fisherDF, log10(ORs) > 2), aes(x = 2, y = -log10(pvalue)), alpha = 0.5,shape=62,size=4) +
  geom_hline(yintercept=-log10(0.05), linetype="dashed", color = "red")+
  annotate(geom = "text",label="p-value = 0.05",x=-2,y=-log10(0.05),hjust=0,vjust=-0.8)

####SPOT####
UbxAllSPOTDisorderedRegions <- getSpotPredDiso(df = yeastDiso_Ubersax_subset,"indices",accession_col = "acc")
UbxAllSPOTStratContingencyArray <- getStratContingencyArray(df = yeastDiso_Ubersax_subset,sequence_col = "sequence",disordered_index_list = UbxAllSPOTDisorderedRegions)

woolf_test(UbxAllSPOTStratContingencyArray)
fisherSpotDF <- data.frame(Ubersax_data$Entry)
colnames(fisherDF) <- "acc"
fisherSpotDF$target <- Ubersax_data$target
fisherSpotDF$ORs <- apply(UbxAllSPOTStratContingencyArray, 3,function(x){
     ft<-fisher.test(x)
     return(ft$estimate)
     })
fisherSpotDF$pvalue <- apply(UbxAllSPOTStratContingencyArray, 3,function(x){
     ft<-fisher.test(x)
     return(ft$p.value)
     })
fisherSpotDF$logPvalue <- -log10(fisherSpotDF$pvalue)
fisherSpotDF$logOr <- log10(fisherSpotDF$ORs)

fisherSpotDF$logOr[is.infinite(fisherSpotDF$logOr) & fisherSpotDF$logOr > 0] <- 2
fisherSpotDF$logOr[is.infinite(fisherSpotDF$logOr) & fisherSpotDF$logOr < 0] <- -2


###Trying to find a way to move the dots out from the logPvalue=0 with random noise(Jitter) in order to asses the amount of points we have there.
set.seed(1234)
fisherSpotDF$logPvalue[fisherSpotDF$logPvalue==0]<- 0.1
fisherSpotDF$logPvalue[fisherSpotDF$logPvalue==0.1]<-abs(jitter(fisherSpotDF$logPvalue[fisherSpotDF$logPvalue==0.1],6))

ggplot(fisherSpotDF, aes(x = logOr, y = logPvalue)) +
  geom_point(aes(color = target)) +
  # scale_color_manual(values = c("", "grey")) +
  theme_bw(base_size = 12) + theme(legend.position = "bottom") +
  scale_y_continuous(trans='log10',limits = c(0.01,10)) +
  xlim(-2, 2) +
  scale_shape_manual()+
  # geom_point(data = subset(fisherDF, log10(ORs) < -2), aes(x = -2, y = -log10(pvalue)), alpha = 0.5,shape=60,size=4) +
  # geom_point(data = subset(fisherDF, log10(ORs) > 2), aes(x = 2, y = -log10(pvalue)), alpha = 0.5,shape=62,size=4) +
  geom_hline(yintercept=-log10(0.05), linetype="dashed", color = "red")+
  annotate(geom = "text",label="p-value = 0.05",x=-2,y=-log10(0.05),hjust=0,vjust=-0.8)



```

```{r agreements disorder vs phosposites2 ,echo=FALSE,message=FALSE,warning=FALSE} 
  
# summary(CDK1targetSacCer$target)
# sd(CDK1targetSacCer$predDisoPercCdkTargets[CDK1targetSacCer$target=="Cdk1 target"])
# sd(CDK1targetSacCer$predDisoPercCdkTargets[CDK1targetSacCer$target=="Non Cdk1 target"])
# var(CDK1targetSacCer$predDisoPercCdkTargets[CDK1targetSacCer$target=="Cdk1 target"])
# var(CDK1targetSacCer$predDisoPercCdkTargets[CDK1targetSacCer$target=="Non Cdk1 target"])
# t.test(x=CDK1targetSacCer$predDisoPercCdkTargets,y=CDK1targetSacCer$target)
# t.test(x=CDK1targetSacCer$predDisoPercCdkTargets[CDK1targetSacCer$target=="Cdk1 target"],y=CDK1targetSacCer$predDisoPercCdkTargets[CDK1targetSacCer$target=="Non Cdk1 target"])
# t.test(x=CDK1targetSacCer$predDisoPercCdkTargets[CDK1targetSacCer$target=="Cdk1 target"],y=CDK1targetSacCer$predDisoPercCdkTargets[CDK1targetSacCer$target=="Non Cdk1 target"])
# t.test(x=CDK1targetSacCer$predDisoPercCdkTargets[CDK1targetSacCer$target=="Cdk1 target"],y=CDK1targetSacCer$predDisoPercCdkTargets[CDK1targetSacCer$target=="Non Cdk1 target"],var.equal = T)
# sd(CDK1targetSacCer$predLiteDisoPercCdkTargets[CDK1targetSacCer$target=="Cdk1 target"])
# sd(CDK1targetSacCer$predLiteDisoPercCdkTargets[CDK1targetSacCer$target=="Non Cdk1 target"])
# CDK1targetSacCer$predLiteDisoPercCdkTargets[CDK1targetSacCer$target=="Non Cdk1 target"]
# sd(CDK1targetSacCer$predLiteDisoPercCdkTargets[CDK1targetSacCer$target=="Non Cdk1 target"],na.rm = T)
# sd(CDK1targetSacCer$predLiteDisoPercCdkTargets[CDK1targetSacCer$target=="Cdk1 target"])
# var(CDK1targetSacCer$predLiteDisoPercCdkTargets[CDK1targetSacCer$target=="Non Cdk1 target"],na.rm = T)
# var(CDK1targetSacCer$predLiteDisoPercCdkTargets[CDK1targetSacCer$target=="Cdk1 target"])
# summary(CDK1targetSacCer$target[!is.na(CDK1targetSacCer$predLiteDisoPercCdkTargets)])
# t.test(x=CDK1targetSacCer$predLiteDisoPercCdkTargets[CDK1targetSacCer$target=="Cdk1 target"],y=CDK1targetSacCer$predLiteDisoPercCdkTargets[CDK1targetSacCer$target=="Non Cdk1 target"],var.equal = T)
# 
# 
# simplePhosphosites<-ggplot(CDK1targetSacCer)+geom_density(aes(x = `[ST]-P`,fill = target),alpha=.5,bw=6)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5)) + scale_y_continuous(limits = c(0,0.1)) + scale_x_continuous(name="sites",limits = c(0,40))+ggtitle("[ST]-P")
# fullPhosphosites<-ggplot(CDK1targetSacCer)+geom_density(aes(x = `[ST]-P-X-[KR]`,fill = target),alpha=.5,bw=4)+ theme_linedraw()+theme(plot.title = element_text(hjust = 0.5)) + theme(axis.title.y = element_blank())+ scale_y_continuous(limits = c(0,0.1)) + scale_x_continuous(name="sites",limits = c(0,40)) + theme(axis.title.y = element_blank())+labs(title="[ST]-P-X-[KR]")
# ggarrange(simplePhosphosites,fullPhosphosites,nrow = 1, ncol = 2,common.legend = TRUE,legend = "bottom")



```


```{r Holt paper data cleansing ,echo=FALSE,message=FALSE,warning=FALSE}



# 
# #Here I'll select only the phosphosites which are positives under the autors considerations.  
# Holt_MSdata_phosphosites <- Holt_MSdata %>% dplyr::filter(grepl("[st]P",centered_15mer)) %>% dplyr::filter(`log2H/L_asynchronous`<(-1)|`log2H/L_nocodazole`<(-1)|`log2H/L_clb2`<(-1))
# 
# 
# #select all the phosphosites from the CDK targets (genes with at least one phosphosite experimentallly detected)
# 
# cdkTargetList <- unique(Holt_MSdata_phosphosites$Uniprot)
# nonCdkTargetList <- unique(Holt_MSdata[!(Holt_MSdata$Uniprot %in% Holt_MSdata_phosphosites$Uniprot),]$Uniprot)
# Holt_CDK1targetSacCer <- Holt_MSdata[Holt_MSdata$Uniprot %in% cdkTargetList,]
#  
# yeastDiso_Holt_CDK1target<-yeastDiso[yeastDiso$acc %in% cdkTargetList,]
# 
# 
# # compare % of disorder
# Holt_proteinInfo <- as.data.frame(rbind(cbind(as.character(cdkTargetList),rep("Cdk1 target",length(cdkTargetList))),cbind(as.character(nonCdkTargetList),rep("Non Cdk1 target",length(nonCdkTargetList)))))
# 
# colnames(Holt_proteinInfo) <- c("Uniprot","cdk1_target")
# Holt_proteinInfo$Uniprot <- as.character(Holt_proteinInfo$Uniprot)
# 
# Holt_proteinInfo<-merge.data.frame(Holt_proteinInfo,yeastDiso,by.x = "Uniprot",by.y = "acc")
# 
# Holt_predDisoPerc <- c()
# for (i in 1:nrow(Holt_proteinInfo)) {
#   mat<-Holt_proteinInfo$disorder$predictors[[i]]$regions[[1]]
#   #calculate the number of disoredered AAs
#   nDiso<-sum(as.numeric(mat[,2])-as.numeric(mat[,1])+1)
#   proteinLength <- nchar(Holt_proteinInfo$sequence[i])
#   Holt_predDisoPerc <- c(Holt_predDisoPerc,(nDiso/proteinLength)*100)
# }
# 
# Holt_proteinInfo$predSimpleDisoPerc <- Holt_predDisoPerc
# 
# 
# Holt_predLiteDisoPerc <- c()
# for (i in 1:nrow(Holt_proteinInfo)) {
#   liteIndex<-which(Holt_proteinInfo$disorder$predictors[[i]]$method=="mobidb-lite")
#   if (length(liteIndex)!=0) {
#     mat<-Holt_proteinInfo$disorder$predictors[[i]]$regions[[liteIndex]]
#     if (class(mat)=="matrix") {
#       #getting only the disordered rows. Is really important to reconvert to a matrix of ncol=3 just in case I index only one row and I get a vector. the the subindex mat[,2] will throw incorrect number of dimensions
#       # mat<-matrix(mat[mat[,3]=="D",],ncol=3)
#       #calculate the number of disoredered AAs
#       nDiso<-sum(as.numeric(mat[,2])-as.numeric(mat[,1])+1)
#       proteinLength <- nchar(Holt_proteinInfo$sequence[i])
#       Holt_predLiteDisoPerc <- c(Holt_predLiteDisoPerc,(nDiso/proteinLength)*100)      
#     } else {Holt_predLiteDisoPerc <- c(Holt_predLiteDisoPerc,0)}
#     
#   } else {
#     Holt_predLiteDisoPerc <- c(Holt_predLiteDisoPerc,NA)
#   }
#   
# }
# Holt_proteinInfo$predLiteDisoPerc <- Holt_predLiteDisoPerc
# 
# 
# #plots
# ggplot(Holt_proteinInfo)+geom_density(aes(x = predSimpleDisoPerc,fill = cdk1_target),alpha=.5,position="identity",bw=5)
# ggplot(Holt_proteinInfo)+geom_density(aes(x = predLiteDisoPerc,fill = cdk1_target),alpha=.5,position="identity",bw=5)
# 
# 
# #Var and SD cheking. t-tests
# var(Holt_proteinInfo$predSimpleDisoPerc[Holt_proteinInfo$cdk1_target=="Non Cdk1 target"])
# var(Holt_proteinInfo$predSimpleDisoPerc[Holt_proteinInfo$cdk1_target=="Cdk1 target"])
# sd(Holt_proteinInfo$predSimpleDisoPerc[Holt_proteinInfo$cdk1_target=="Cdk1 target"])
# sd(Holt_proteinInfo$predSimpleDisoPerc[Holt_proteinInfo$cdk1_target=="Non Cdk1 target"])
# 
# t.test(Holt_proteinInfo$predSimpleDisoPerc[Holt_proteinInfo$cdk1_target=="Cdk1 target"],Holt_proteinInfo$predSimpleDisoPerc[Holt_proteinInfo$cdk1_target=="Non Cdk1 target"],var.equal=T)
# 
# 
# 
# var(Holt_proteinInfo$predLiteDisoPerc[Holt_proteinInfo$cdk1_target=="Cdk1 target"],na.rm = T)
# var(Holt_proteinInfo$predLiteDisoPerc[Holt_proteinInfo$cdk1_target=="Non Cdk1 target"],na.rm = T)
# sd(Holt_proteinInfo$predLiteDisoPerc[Holt_proteinInfo$cdk1_target=="Cdk1 target"],na.rm = T)
# sd(Holt_proteinInfo$predLiteDisoPerc[Holt_proteinInfo$cdk1_target=="Non Cdk1 target"],na.rm = T)
# 
# t.test(Holt_proteinInfo$predLiteDisoPerc[Holt_proteinInfo$cdk1_target=="Cdk1 target"],Holt_proteinInfo$predLiteDisoPerc[Holt_proteinInfo$cdk1_target=="Non Cdk1 target"],var.equal=T)
# 
# ########SPOT#############
# Holt_predSpotDisoPerc<-c()
# for (i in 1:nrow(Holt_proteinInfo)) {
#   diso<-yeastDiso[yeastDiso$acc==Holt_proteinInfo$Uniprot[[i]],"predSpotDisoPercAll"]
#   Holt_predSpotDisoPerc<-c(Holt_predSpotDisoPerc,diso)
# }
# Holt_proteinInfo$predSpotDisoPerc <- Holt_predSpotDisoPerc
# 
# ggplot(Holt_proteinInfo)+geom_density(aes(x = predSpotDisoPerc,fill = cdk1_target),alpha=.5,position="identity",bw=5)
# 
# 
# #Var and SD cheking. t-tests
# var(Holt_proteinInfo$predSpotDisoPerc[Holt_proteinInfo$cdk1_target=="Non Cdk1 target"])
# var(Holt_proteinInfo$predSpotDisoPerc[Holt_proteinInfo$cdk1_target=="Cdk1 target"])
# sd(Holt_proteinInfo$predSpotDisoPerc[Holt_proteinInfo$cdk1_target=="Cdk1 target"])
# sd(Holt_proteinInfo$predSpotDisoPerc[Holt_proteinInfo$cdk1_target=="Non Cdk1 target"])
# 
# t.test(Holt_proteinInfo$predSpotDisoPerc[Holt_proteinInfo$cdk1_target=="Cdk1 target"],Holt_proteinInfo$predSpotDisoPerc[Holt_proteinInfo$cdk1_target=="Non Cdk1 target"],var.equal=T)

```

Holt paper

- between the cdk1 targets and non cdk1 targets we've got 1977 proteins with detectable peptides out of 6726 in the Uniprot yeast proteome.
- The peaks were selected with the consideration of A-score > 13. They don't mention to be acounting for neutral loss

```{r Compare papers,echo=FALSE,message=FALSE,warning=FALSE}


# sets <- list(Ubersax_cdkTargets=CDK1targetSacCer$Entry[CDK1targetSacCer$`P-Score`>=2],
#              Ubersax_nonCdkTargets=CDK1targetSacCer$Entry[CDK1targetSacCer$`P-Score`<2],
#              Holt_cdkTargets=cdkTargetList,
#              Holt_nonCdkTargets=nonCdkTargetList)
# 
# eulerr_options(fills=list(fill=c("darkolivegreen1","deeppink3","darkcyan","darkorange1"),alpha=0.4))
# plot(euler(sets, shape = "ellipse"), quantities = TRUE)


```
## Sequence characteristics of disorder regions in proteins

Considering the analysis performed previously, It seems evident that disorder regions are enriched in certain residues while they presents less proportion of others. Serine, Threonine, Proline Asparagine and charged amino acids are present in a bigger proportion in disorder regions when we compare with the compositions of the entire proteins. Knowing the sequence motif for the minimal consensus phosphosite, the first question that raises is if the higher propensity of CDK1 phosphosites falling into disordered regions is due only by the fact that predictors recognize Ser, Pro and Thr rich regions as IDR.  

<a href="supplementary/Supp.html">Supplementary</a>

```{r save_env,echo=FALSE,message=FALSE,warning=FALSE}
save.image("~/projects/cdk1_cyclinB1/phase_separation/disorder/disoWS.RData")
```

# References  
  
